ddtrace/tracer/servicehash.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package tracer
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"sync"
+)
+
+const maxCacheSize = 1000
+
+var (
+	cache = make(map[string]string)
+	lock  sync.RWMutex
+)
+
+// Hash computes hash for `service`. Will use a local cache to avoid recomputation.
+func Hash(service string) string {
+	hash := getHashFromCache(service)
+	if len(hash) > 0 {
+		return hash
+	}
+	hashb := sha256.Sum256([]byte(service))
+	// Only grab first 10 characters (2 chars per byte * 5 bytes = 10 chars)
+	hash = hex.EncodeToString(hashb[:5])
+	setHashInCache(service, hash)
+	return hash
+}
+
+func getHashFromCache(service string) string {
+	lock.RLock()
+	defer lock.RUnlock()
+	if hash, ok := cache[service]; ok {
+		return hash
+	}
+	return ""
+}
+
+func setHashInCache(service, hash string) {
+	lock.Lock()
+	defer lock.Unlock()
+	if len(cache) == maxCacheSize {
+		// deletes a random key
```

Comment:
The efficiency of it will be random too, once it gets full. Not sure if the edge case is worth it, but I'd implement a proper LRU cache, something like: https://github.com/golang/go/blob/2882786bf4cd779f166e9ced82a4da2ea0f8b1f9/src/net/http/transport.go#L2868-L2871

I don't feel strongly about this, so if you can't afford the time or effort, I'm ok with merging as is since the gain of this is very on the edge case side.

Reply:
Yeah this is actually a question I have, it seems like the vast majority of go programs wouldn't have more than 1k unique service names? Or are there typical customer use-cases that could result in filling this cache regularly?

(Personally I like the simplicity here by avoiding LRU logic, but you are definitely right that when the cache is full the performance will suffer)

Reply:
Yeah, given the amount of times this may get called, I think that optimising here might be overkill. Resolving.